File: ContosoAir.Data/Entities/Flight.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace ContosoAir.Data.Entities
{
    public class Flight
    {
        public int Id { get; set; }
        public int Price { get; set; }
        public string Duration { get; set; }
        public string FromCode { get; set; }
        public string ToCode { get; set; }
        public int Distance { get; set; }
        public List<FlightSegment> Segments { get; set; }
    }
}

File: ContosoAir.Site/src/app/flights/flight.ts
export interface Flight {
    id: number,
    duration: string,
    price: string,
    fromCode: string,
    toCode: string,
    distance: string,
    segments: {
    	flight: string,
    	fromCode: string,
    	departTime: string,
    	toCode: string,
    	arrivalTime: string
    }[]
}